bin/locale-generator/locale-generator.php

```diff
@@ -0,0 +1,146 @@
+					$locales->locales[ $key ]->cldr_plural_expressions['many'] = $rule;
+
+					break;
+				case 'pluralRule-count-other':
+					$locales->locales[ $key ]->cldr_plural_expressions['other'] = $rule;
+
+					break;
+			}
+		}
+	}
+}
+
+// Use the new GP Locales object to create output that can be added in to the locales.php file.
+generate_output( $locales );
+/**
+ * Function to output the locales data in PHP source format.
+ *
+ * @param GP_Locales $locales The GlotPress Locales object.
+ */
+function generate_output( $locales ) {
+	// Get all the GP_Locale variables we're going to output.
+	$vars = get_object_vars( $locales->locales['en'] );
+
+	// Loop through all of the locales.
+	foreach ( $locales->locales as $locale ) {
+		// Create the variable name we'll use to define the locale based on the slug with any dashes replaced with underscores.
+		$var_name = str_replace( '-', '_', $locale->slug );
+		$root_var_name = str_replace( '-', '_', $locale->variant_root );
+
+		// Output the first line to 'create' the GP_Locale object for this locale.
+		echo "\t\t\${$var_name} = new GP_Locale();\n"; // WPCS: XSS ok.
+
+		// Now loop through all the variables that may be set for this locale.
+		foreach ( $vars as $var => $value ) {
+			// Handle variables that are arrays for output.
+			if ( is_array( $locale->$var ) ) {
+				// Loop through all of the array keys to output.
+				foreach ( $locale->$var as $key => $value ) {
+					// Don't output empty variables or the speical case of 'variants' (they will be outputed with the variant locale instead).
+					if ( ! empty( $locale->$var[ $key ] ) && 'variants' !== $var ) {
+						// Create some space to make the output pretty and they output the line.
+						$padding = str_repeat( ' ', 36 - strlen( $var ) - ( strlen( $var_name ) - 2 ) - strlen( $key ) - 4 );
+						echo "\t\t\${$var_name}->{$var}['{$key}']{$padding} = '" . str_replace( "'", "\'", $locale->$var[ $key ] ) . "';\n"; // WPCS: XSS ok.
+					}
+				}
+			} else {
+				// Don't output empty variables.
+				if ( ! empty( $locale->$var ) ) {
+					// Create some space to make the output pretty and they output the line.
+					$padding = str_repeat( ' ', 36 - strlen( $var ) - ( strlen( $var_name ) - 2 ) );
+					echo "\t\t\${$var_name}->{$var}{$padding} = '" . str_replace( "'", "\'", $locale->$var ) . "';\n"; // WPCS: XSS ok.
+
+					// Handle the special case of locales with variants, output the line to create the root's variant list here.
+					if ( 'variant_root' === $var ) {
+						// Create some space to make the output pretty and they output the line.
+						$padding = str_repeat( ' ', 36 - strlen( $root_var_name ) - ( strlen( $root_var_name ) - 2 ) - 15 );
+						echo "\t\t\${$root_var_name}->variants['{$var_name}']{$padding} = \${$root_var_name}->english_name;\n"; // WPCS: XSS ok.
+					}
+				}
+			}
+		} // End foreach().
+
+		// Add some space between locales for easier reading.
+		echo "\n"; // WPCS: XSS ok.
```

Comment:
Can we ignore the variants part in this PR for now?

Reply:
Not easily without breaking one or both of the patches :)

They both do a lot of work on the locales file so separating them is hard.